### PR TITLE
Muen subject info API update

### PR DESCRIPTION
**Pull request by Kensan**
The subject info API was recently [restructured](https://git.codelabs.ch/?p=muen.git;a=commit;h=a4e69e2). Adapt all Muen drivers to work with the new Sinfo API.

### CODE REVIEW

File: kernel/muen/muen-net.c [hunk 1]
```diff
@@ -106,29 +106,53 @@ void generate_mac_addr(uint8_t *addr)
...
         log(WARN, "Solo5: Net: No output channel\n");
         return;
     }
-    net_out  = (struct muchannel *)(channel.address);
+    if (!(chan_out->data.mem.flags & MEM_CHANNEL_FLAG)) {
+        log(WARN, "Solo5: Net: Memory '%s' is not a channel\n",
+            chan_out->name.data);
+        return;
+    }
+	if (!(chan_out->data.mem.flags & MEM_WRITABLE_FLAG)) {
```
Comment (mato) [inline kernel/muen/muen-net.c]: Nit: Whitespace/indent is wrong here.

File: kernel/muen/muen-net.c [hunk 2]
```diff
@@ -106,29 +106,53 @@ void generate_mac_addr(uint8_t *addr)
...
         log(WARN, "Solo5: Net: No input channel\n");
         return;
     }
-    net_in = (struct muchannel *)(channel.address);
+    if (!(chan_in->data.mem.flags & MEM_CHANNEL_FLAG)) {
+        log(WARN, "Solo5: Net: Memory '%s' is not a channel\n",
+            chan_in->name.data);
+        return;
+    }
+	if (chan_in->data.mem.flags & MEM_WRITABLE_FLAG) {
```
Comment (mato) [inline kernel/muen/muen-net.c]: As above, indent seems wrong.

**Comment by mato** [comment]
@Kensan This looks fine, you mentioned to me that it's a API breaking change; do you want me to merge this now or wait until you've released(?) Muen with this change? Either is fine from the Solo5 side as there is no formal release with Muen support yet.

**Comment by Kensan** [comment]
> Do you want me to merge this now or wait until you've released(?) Muen with this change?

Since the changes are already in the Muen devel branch I think it would be good if you can merge these changes now.

**Pull request merged**